### PR TITLE
redirect shorts view to "watch" view, block paths like /feed /gaming ...

### PR DESCRIPTION
**Pull request by thuan00**
hi,
i really appreciate the work

i noticed shorts are still accessible with all the infinite scroll and also other pages like /feed/trending or /gaming
so i proposed these changes that work for me

have a look and let me know what you think

--
*update 1:
the last rule of redirecting to home seems a bit strict, but i think its ok
i added permission for `/playlist` and `/redirect`
and i checked out all the paths i can reach so nothing essential is blocked and redirected to home

these are some paths that can be distracting:
```
/reporthistory
/feed/history
/feed/trending
/feed/subscriptions
/gaming
```

these are misc:
```
/account
/about
/intl
/t
/ads
/creators
...
```

more paths can be seen here, tho i think they are not too relevant
https://www.youtube.com/sitemaps/sitemap.xml

--
*update 2:
commit 4
youtube is complicated and i probably dont know all the paths and apis
so i make a blacklist instead of the strict rule, redirecting all unspecified paths to home
the `/embed` path is essential

### CODE REVIEW

File: src/extension.js [hunk 1]
```diff
@@ -37,19 +37,45 @@ const enableTheaterMode = () => {
   document.cookie = "wide=1; expires="+oneYearFromNow.toUTCString()+"; path=/; domain=.youtube.com"
 }
 
+const pathBlacklist = [
+  "/feed",
+  "/gaming",
+  "/reporthistory",
+]
+const isBlacklistedPath = (path) => {
```
Comment (makaroni4) [inline src/extension.js:45]: Hey @thuan00 👋 

Thanks a ton for a PR 🙌  

Let's see what we have here.

File: src/extension.js [hunk 2]
```diff
@@ -37,19 +37,45 @@ const enableTheaterMode = () => {
...
-  } else if (window.location.pathname === "/results") {
+  } else if (pathname === "/results") {
     initResultsPage()
-  } else if (window.location.pathname === "/watch") {
+  } else if (pathname === "/watch") {
     initWatchPage()
-  } else if (window.location.pathname.startsWith("/@") || window.location.pathname.startsWith("/channel")) {  // channel begins with /@ or /channel
+  } else if (pathname.startsWith("/@") || pathname.startsWith("/channel")) {  // channel begins with /@ or /channel
+    initChannelPage()
+  } else if (pathname === "/playlist") {
```
Comment (makaroni4) [inline src/extension.js:69]: 🔍 Good news: we now handle the playlist page properly https://github.com/makaroni4/focused_youtube/pull/36/files

File: src/extension.js [hunk 3]
```diff
@@ -37,19 +37,45 @@ const enableTheaterMode = () => {
...
+    // temporarily using the channel page to hide the side bar
     initChannelPage()
+  } else if (pathname.startsWith("/shorts")) {
```
Comment (makaroni4) [inline src/extension.js:72]: 🔍 I think it's the same situation as with the blacklist – ideally, we've blocked all shorts from pages, so one can't simply access them (and stays focused 💪 😄 ).

File: src/extension.js [hunk 4]
```diff
@@ -37,19 +37,45 @@ const enableTheaterMode = () => {
...
+  return false
+}
+
 const initFY = () => {
   cleanUpFYClasses()
 
   enableTheaterMode()
 
-  if (window.location.pathname === "/") {
+  const pathname = window.location.pathname
```
Comment (makaroni4) [inline src/extension.js:59]: 🔍 I like that DRY refactoring, let's clean up the rest and merge it 🧹

**Comment by thuan00** [comment]
hi @makaroni4 ,

thanks for the review

i had to be strict using the path approach because of a few reasons:
- i some times find some helpful tutorial shorts from google search, and then i have all the infinite scroll with distracting content
- i can type "shorts" directly to the url bar to watch shorts when im bored, the same for "gaming" and "feed"
- some shorts appear under the description, the remix section (for example: https://www.youtube.com/watch?v=tzbLDH7Wa5o)

it is not necessary to get this merged, so feel free to close or edit my PR
maybe it is just to give you an idea on how i solve my personal problem

**Comment by makaroni4** [comment]
Roger, let's skip it for now then @thuan00 🍻 

💡 You can disable the infinite scroll in the extension config btw:

Screenshot 2024-09-02 at 14 13 25

**Pull request closed without merging**